src/main/kotlin/com/sourcegraph/cody/agent/protocol/ContextFile.kt

```diff
@@ -1,7 +1,26 @@
 package com.sourcegraph.cody.agent.protocol
 
+import com.google.gson.JsonDeserializationContext
+import com.google.gson.JsonDeserializer
+import com.google.gson.JsonElement
+import java.lang.reflect.Type
+import java.net.URI
+
 data class ContextFile(
-    val fileName: String,
+    val uri: URI,
     val repoName: String?,
     val revision: String?,
 )
```

Comment:
We should test this String->URI change on Windows. Recently, there were issues because the agent has a URI in a different RFC than those built-in in Java.